x-pack/plugins/apm/public/components/app/correlations/correlations_chart.tsx

```diff
@@ -0,0 +1,229 @@
+  AnnotationDomainType,
+  Chart,
+  CurveType,
+  Settings,
+  Axis,
+  ScaleType,
+  Position,
+  AreaSeries,
+  RecursivePartial,
+  AxisStyle,
+  PartialTheme,
+  LineAnnotation,
+  LineAnnotationDatum,
+} from '@elastic/charts';
+
+import euiVars from '@elastic/eui/dist/eui_theme_light.json';
+
+import { EuiSpacer } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { getDurationFormatter } from '../../../../common/utils/formatters';
+
+import { useTheme } from '../../../hooks/use_theme';
+import { HistogramItem } from '../../../../common/search_strategies/correlations/types';
+
+const { euiColorMediumShade } = euiVars;
+const axisColor = euiColorMediumShade;
+
+const axes: RecursivePartial<AxisStyle> = {
+  axisLine: {
+    stroke: axisColor,
+  },
+  tickLabel: {
+    fontSize: 10,
+    fill: axisColor,
+    padding: 0,
+  },
+  tickLine: {
+    stroke: axisColor,
+    size: 5,
+  },
+  gridLine: {
+    horizontal: {
+      dash: [1, 2],
+    },
+    vertical: {
+      strokeWidth: 1,
+    },
+  },
+};
+const chartTheme: PartialTheme = {
+  axes,
+  legend: {
+    spacingBuffer: 100,
+  },
+  areaSeriesStyle: {
+    line: {
+      visible: false,
+    },
+  },
+};
+
+// Default axis to start at 0 for latency correlations chart
+const xAxisDomain = {
+  min: 0,
+};
+
+interface CorrelationsChartProps {
+  field?: string;
+  value?: string;
+  histogram?: HistogramItem[];
+  markerValue: number;
+  markerPercentile: number;
+  overallHistogram: HistogramItem[];
+}
+
+const annotationsStyle = {
+  line: {
+    strokeWidth: 1,
+    stroke: 'gray',
+    opacity: 0.8,
+  },
+  details: {
+    fontSize: 8,
+    fontFamily: 'Arial',
+    fontStyle: 'normal',
+    fill: 'gray',
+    padding: 0,
+  },
+};
+
+const CHART_PLACEHOLDER_VALUE = 0.0001;
+
+// Elastic charts will show any lone bin (i.e. a populated bin followed by empty bin)
+// as a circular marker instead of a bar
+// This provides a workaround by making the next bin not empty
+export const replaceHistogramDotsWithBars = (
+  originalHistogram: HistogramItem[] | undefined
+) => {
+  if (originalHistogram === undefined) return;
+  const histogram = [...originalHistogram];
+  {
+    for (let i = 0; i < histogram.length - 1; i++) {
+      if (
+        histogram[i].doc_count > 0 &&
+        histogram[i].doc_count !== CHART_PLACEHOLDER_VALUE &&
+        histogram[i + 1].doc_count === 0
+      ) {
+        histogram[i + 1].doc_count = CHART_PLACEHOLDER_VALUE;
+      }
+    }
+    return histogram;
+  }
+};
+
+export function CorrelationsChart({
+  field,
+  value,
+  histogram: originalHistogram,
+  markerValue,
+  markerPercentile,
+  overallHistogram,
+}: CorrelationsChartProps) {
+  const euiTheme = useTheme();
+
+  if (!Array.isArray(overallHistogram)) return <div />;
+  const annotationsDataValues: LineAnnotationDatum[] = [
+    { dataValue: markerValue, details: `${markerPercentile}th percentile` },
+  ];
+
+  const xMax = max(overallHistogram.map((d) => d.key)) ?? 0;
```

Comment:
I don't see any benefits of this replacement as APM already imports/uses `d3.max`

Reply:
I originally used `d3.min/max` because in pre ES6 times this was an easier way since it always allowed arrays. With the pattern suggested by Dima we could get rid of a bit dependency code and smaller bundle size.

Reply:
Updated to es6 max here https://github.com/elastic/kibana/pull/99905/commits/78ab41163f8a08c36670dbcabe82186d1c8dfa00